http/server.ts

```diff
@@ -647,6 +738,9 @@ export async function listenAndServe(
  * @param keyFile The path to the file containing the TLS private key.
  * @param handler The handler for individual HTTP requests.
  * @param options Optional serve options.
+ *
+ * @deprecated Use `serve` instead. `serve` supports `tls` options, which starts the server with
+ * TLS connections.
```

Comment:
Rather than overloading server to handle both TCP and TLS, why not add a `serveTls` function? I would find that clearer...

Reply:
Ok. Let's do that. Updated